AWSClientRuntime/Sources/Auth/AWSCredentialsProviderSTSConfig.swift

```diff
@@ -7,64 +7,26 @@
         
 import AwsCommonRuntimeKit
 import ClientRuntime
+import Foundation
         
 public struct AWSCredentialsProviderSTSConfig {
     public let credentialsProvider: AWSCredentialsProvider
     public let roleArn: String
     public let sessionName: String
-    public let durationSeconds: UInt16
+    public let durationSeconds: TimeInterval
     public let shutDownCallback: ShutDownCallback?
-    private let sharedDefaultIO: SDKDefaultIO = SDKDefaultIO.shared
     
-    public init(credentialsProvider: AWSCredentialsProvider,
-                roleArn: String,
-                sessionName: String,
-                durationSeconds: UInt16,
-                shutDownCallback: ShutDownCallback? = nil) {
+    public init(
+        credentialsProvider: AWSCredentialsProvider,
+        roleArn: String,
+        sessionName: String,
+        durationSeconds: TimeInterval,
+        shutDownCallback: ShutDownCallback? = nil
+    ) throws {
         self.credentialsProvider = credentialsProvider
         self.roleArn = roleArn
         self.sessionName = sessionName
         self.durationSeconds = durationSeconds
         self.shutDownCallback = shutDownCallback
     }
 }
-
-extension AWSCredentialsProviderSTSConfig {
-    func toCRTType() -> CredentialsProviderSTSConfig {
-        return CredentialsProviderSTSConfig(bootstrap: SDKDefaultIO.shared.clientBootstrap,
-                                            tlsContext: SDKDefaultIO.shared.tlsContext,
-                                            credentialsProvider: credentialsProvider.crtCredentialsProvider,
-                                            roleArn: roleArn,
-                                            sessionName: sessionName,
-                                            durationSeconds: durationSeconds,
-                                            shutDownOptions: shutDownCallback)
-    }
-}
-
-struct CredentialsProviderSTSConfig: CRTCredentialsProviderSTSConfig {
```

Comment:
not required?

Reply:
Nope. This struct no longer has any purpose. 
Checkout the related factory method in [AwsCredentialsProvider]](https://github.com/awslabs/aws-sdk-swift/pull/777/files#diff-c858c06a42a458b91bb72c1ce5190f515a40cf4463775e74d8f8f91d4e53c506) to see why its no needed.